src/serialbox-fortran/CMakeLists.txt

```diff
@@ -10,19 +10,22 @@
 cmake_minimum_required(VERSION 3.12)
 
 if(SERIALBOX_ENABLE_FORTRAN)
-
-  set(SOURCES_FORTRAN_SERIALIZE m_serialize.f90 m_ser_perturb.f90)
+  
+  set(SOURCES_FORTRAN_SERIALIZE m_serialize.f90 m_ser_perturb.f90 m_ser_kbuffer.f90 utils_ppser.f90)
   add_library(SerialboxFortranSerializeObjects OBJECT ${SOURCES_FORTRAN_SERIALIZE})
   target_include_directories(SerialboxFortranSerializeObjects
     PUBLIC
         $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>
         $<INSTALL_INTERFACE:include>
     )
-  
-  set(SOURCES_FORTRAN utils_ppser.f90 )

```

Comment:
Moving `utils_ppser.f90` in the same target as `m_serialize.f90` will introduce a race as the m_serialize module might not be available when compiling utils_ppser.f90.

Reply:
I reverted the change.

Reply:
Actually, moving it out did not work. CMake has automatic Fortran dependency detection within the same target and thus the files are compiled in the correct order.

Reply:
Took me quite some time to recover under which conditions this is a race condition. Indeed in the current setup it is not. The problem arises when the src file containing a module is used in several targets. I think in an earlier version of this CMake file, we had `add_library(STATIC <all .f90 files>)` and `add_library(SHARED <all .f90 files>)` which probably created the race.
Do you remember when we tried to build COSMO with CMake? We run into a similar issue and I wonder now if this was just because yoda was building shared and static at the same time by default.

Reply:
Actually, it didn't work work because you did 

```
  set(SOURCES_FORTRAN utils_ppser.f90)
  set(SOURCES_FORTRAN utils_ppser_kbuff.f90) # <-- overwrites the variable
```
instead of 
```
  set(SOURCES_FORTRAN utils_ppser.f90 utils_ppser_kbuff.f90)
```

Reply:
Change done in #222 .